src/index.js

```diff
@@ -2,21 +2,27 @@ const core = require('@actions/core');
 const github = require('@actions/github');
 
 async function main() {
-  const { eventName, sha, ref, repo: { owner, repo } } = github.context;
-  const branch = ref.slice(11);
-  console.log({ eventName, sha, branch, owner, repo });
+  const { eventName, sha, ref, repo: { owner, repo }, payload } = github.context;
+  let branch = ref.slice(11);
+  let headSha = sha;
+  if (payload.pull_request) {
+    branch = payload.pull_request.head.ref;
+    headSha = payload.pull_request.head.sha;
+  }
+
+  console.log({ eventName, sha, headSha, branch, owner, repo, branch });
   const workflow_id = core.getInput('workflow_id', { required: true });
   const token = core.getInput('access_token', { required: true });
   console.log(`Found input: ${workflow_id}`);
   console.log(`Found token: ${token ? 'yes' : 'no'}`);
   const octokit = new github.GitHub(token);
-  const { data } = await octokit.actions.listWorkflowRuns({ owner, repo, workflow_id });
+  const { data } = await octokit.actions.listWorkflowRuns({ owner, repo, workflow_id, branch });
```

Comment:
Amazing! I must have missed the option for `branch` before 🥇

Reply:
Yes, we can also give the status, but as we both want `in_progress` and `queued` it wouldn't work for us